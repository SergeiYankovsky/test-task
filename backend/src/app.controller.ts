import { Controller, Get } from '@nestjs/common';
import { AppService } from './app.service';
import { ApiOkResponse, ApiTags } from '@nestjs/swagger';

@Controller({
  version: '1',
})
@ApiTags('ping')
export class AppController {
  constructor(private readonly appService: AppService) { }

  @Get('ping')
  @ApiOkResponse({ description: 'Return "pong" if server is running' })
  ping(): string {
    return this.appService.ping();
  }
}
