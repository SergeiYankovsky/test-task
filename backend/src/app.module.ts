import { Module } from '@nestjs/common';
import { AppController } from './app.controller';
import { AppService } from './app.service';
import { ContactsModule } from './modules/contacts/contacts.module';
import { TypeOrmModule } from '@nestjs/typeorm';
import { ConfigModule } from '@nestjs/config';
import { Contact } from './modules/contacts/entities/contact.entity';
import { APP_FILTER } from '@nestjs/core';
import { HttpErrorFilter } from './errors/http-error.filter';

@Module({
  imports: [
    ContactsModule,
    ConfigModule.forRoot(),
    TypeOrmModule.forRoot({
      type: 'postgres',
      host: process.env.DB_HOST,
      port: +process.env.DB_PORT,
      username: process.env.DB_USERNAME,
      password: process.env.DB_PASSWORD,
      database: process.env.DB_NAME,
      entities: [Contact],
      synchronize: true,
    }),
  ],
  controllers: [AppController],
  providers: [
    AppService,
    {
      provide: APP_FILTER,
      useClass: HttpErrorFilter,
    }
  ],
})
export class AppModule { }
