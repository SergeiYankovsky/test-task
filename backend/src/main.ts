import { NestFactory } from "@nestjs/core";
import { AppModule } from "./app.module";
import { ValidationPipe, VersioningType } from "@nestjs/common";
import { DocumentBuilder, SwaggerModule } from "@nestjs/swagger";
import { GLOBAL_PREFIX } from "./constants";

declare global {
  namespace NodeJS {
    interface ProcessEnv {
      DB_HOST: string;
      DB_PORT: number;
      DB_USERNAME: string;
      DB_PASSWORD: string;
      DB_NAME: string;
      NODE_ENV: "dev" | "prod";
      PORT: number;
    }
  }
}

async function bootstrap() {
  const app = await NestFactory.create(AppModule, {
    cors: true,
  });

  app.setGlobalPrefix(GLOBAL_PREFIX);
  app.useGlobalPipes(
    new ValidationPipe({
      whitelist: true,
    })
  );
  app.enableVersioning({
    type: VersioningType.URI,
  });

  const config = new DocumentBuilder()
    .setTitle("Contacts example")
    .setDescription("The contacts API description")
    .setVersion("1.0")
    .addTag("contacts")
    .build();
  const document = SwaggerModule.createDocument(app, config);
  SwaggerModule.setup("api", app, document);

  const port = process.env.PORT || 3000;

  await app.listen(port, async () => {
    const url = await app.getUrl();

    console.log(`Application is running on: ${url}${GLOBAL_PREFIX}`);
  });
}
bootstrap();
