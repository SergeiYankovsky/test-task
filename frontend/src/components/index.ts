export * from "./Input";
export * from "./Button";
export * from "./Card";
export * from "./Header";
export * from "./Modal";
export * from "./Form";
