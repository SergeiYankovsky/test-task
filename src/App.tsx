import "./styles/global.scss";
import { MainPage } from "./pages/main-page";
import { QueryClientProvider, QueryClient } from "@tanstack/react-query";
const queryClient = new QueryClient();
export const App: React.FC = () => {
  return (
    <QueryClientProvider client={queryClient}>
      <MainPage />
    </QueryClientProvider>
  );
};
