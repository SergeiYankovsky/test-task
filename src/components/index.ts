export * from "./Input";
export * from "./Button";
export * from "./Form";
export * from "./Card";
export * from "./Header";
export * from "./Modal";
