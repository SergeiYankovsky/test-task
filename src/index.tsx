import { Container, createRoot } from "react-dom/client";
import { App } from "./App";
import React from "react";

const container = document.getElementById("root");

const root = createRoot(container as Container);
root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>,
);
