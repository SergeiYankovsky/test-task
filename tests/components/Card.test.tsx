import { Card } from "../../src/components";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Card
        title={""}
        email={""}
        onClick={function (): void {
          throw new Error("Function not implemented.");
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
