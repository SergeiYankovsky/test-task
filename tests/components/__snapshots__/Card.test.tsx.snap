// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="card"
  onClick={[Function]}
>
  <h3
    className="title"
  />
  <p
    className="description"
  />
</button>
`;
